templates/aws-stack.yml

```diff
@@ -818,23 +819,34 @@ Parameters:
 
   InstanceRoleName:
     Type: String
-    Description: Optional - A name for the IAM Role attached to the Instance Profile.
+    Description: Optional - A name for the IAM Role attached to the Instance Profile when creating a new role. Ignored when InstanceRoleARN is provided.
     Default: ""
 
   InstanceRolePermissionsBoundaryARN:
     Type: String
-    Description: Optional - The ARN of the policy used to set the permissions boundary for the role.
+    Description: Optional - The ARN of the policy used to set the permissions boundary for the role when creating a new role. Ignored when InstanceRoleARN is provided.
     Default: ""
 
   InstanceRoleTags:
     Description: >
       Optional - Comma-separated key=value pairs for instance IAM role tags (up to 5 tags).
       Example: 'Environment=production,Team=platform,Purpose=ci'.
       Note: Keys and values cannot contain '=' characters.
+      Only applied when creating a new role, ignored when InstanceRoleARN is provided.
     Type: String
     Default: ""
     AllowedPattern: "^$|^[\\w\\s_.:/+\\-@]+=[\\w\\s_.:/+\\-@]*(,[\\w\\s_.:/+\\-@]+=[\\w\\s_.:/+\\-@]*){0,4}$"
 
+  InstanceRoleARN:
+    Type: String
+    Description: >
+      Optional - ARN of an existing IAM role to attach to instances instead of creating a new role.
+      When specified, the stack will not create any IAM roles or policies, and will use this role instead.
+      The role must have all necessary permissions for Buildkite agents to function correctly.
+      This is useful when you want to share a single IAM role across multiple queues/stacks.
+    Default: ""
+    AllowedPattern: "^$|^arn:aws:iam::[0-9]+:role/.+$"
```

Comment:
This pattern won't match roles with custom paths: `arn:aws:iam::123456789012:role/custom/path/MyRole`